### ISSUE TITLE
Using formula fields to prepopulate created & last modified data

### ISSUE THREAD

**Issue by flatlines** [mdelobelle/metadatamenu#350]
Hello,

I am new to metadata menu and might be doing this wrong.

I have created the following field in my fileClass note:

`created:: {"type":"Formula","options":{"autoUpdate":false,"formula":"current.file.cday.toFormat(\"yyyy-MM-dd\"))"}}`

When I insert a "created::" field in my target note, nothing happens.

From the video [here](https://www.youtube.com/watch?v=LqglkrzLAoQ), I would have thought that the field would pre-populate with the date on which this particular note was created, e.g. 2023-01-02

I have also tried the following variant: `dv.current().file.cday.toFormat("yyyy-MM-dd")` but again, the field is empty on insertion.

Setting the auto-update to "true" does not seem to make a difference.

### Expected behaviour based on video
Once a formula field has been set up in the fileClass, it will auto-populate in any note that has that file-class.

### Actual behaviour
The formula field is empty.

**Note** I have created the formula filed under the _fileClass_ note, not under **Preset Field Settings** in the metadata menu.

**Comment by flatlines** [comment]
Quick update on this, I have a variant working when it is created via the **Preset Field Settings** menu. The correct call is: `current.file.cday.toFormat("yyyy-MM-dd")`

However, when I use the **Add New Field** modal menu in my fileClass template, that same formula _does not work_. What's interesting is that the modal generates the following text:
`current:: {"type":"Formula","options":{"autoUpdate":false,"formula":"current.file.cday.toFormat(\"yyyy-MM-dd\")"}}`

I am guessing the `\` is an escaped character?

Even if I manually remove these escaped characters the formula still doesn't run.

## TL;DR:
The following `created day` (`current.file.cday.toFormat("yyyy-MM-dd")`) formula _works_ when it is added via the **preset field settings** menu but will then break reliably if added inside a fileClass template.

**Comment by flatlines** [comment]
Final update as a result of further testing. I have now gotten it to work under the _fileClass_ note, but the added field has to be right-clicked + manually prompted to "update formula" using the menu. Running the update formula command on the entire page doesn't seem to do it. Sometimes the field contents will flash and disappear (not sure why). Manually typing in the field key, e.g. "current::" doesn't seem to be recognised as valid by the system.

**Comment by flatlines** [comment]
Final update - setting last_modified (`current.file.mday.toFormat("yyyy-MM-dd")`) to auto-update true crashes obsidian.

**Comment by technotherion** [comment]
Whoops, wish I would have seen this thread before I embarked on the same journey.
I experienced the same crash with a similar command and described it in further detail here: https://github.com/mdelobelle/metadatamenu/issues/370
